ci/gpu/build.sh

```diff
@@ -111,13 +111,18 @@ else
     gpuci_logger "Running dask.distributed GPU tests"
     # Test downstream packages, which requires Python v3.7
     if [ $(python -c "import sys; print(sys.version_info[1])") -ge "7" ]; then
-        gpuci_logger "TEST OF DASK/UCX"
-        py.test --cache-clear -vs `python -c "import distributed.protocol.tests.test_cupy as m;print(m.__file__)"`
-        py.test --cache-clear -vs `python -c "import distributed.protocol.tests.test_numba as m;print(m.__file__)"`
-        py.test --cache-clear -vs `python -c "import distributed.protocol.tests.test_rmm as m;print(m.__file__)"`
-        py.test --cache-clear -vs `python -c "import distributed.protocol.tests.test_collection_cuda as m;print(m.__file__)"`
-        py.test --cache-clear -vs `python -c "import distributed.tests.test_nanny as m;print(m.__file__)"`
-        py.test --cache-clear -vs `python -c "import distributed.diagnostics.tests.test_nvml as m;print(m.__file__)"`
+        # Clone Distributed to avoid pytest cleanup fixture errors
+        # See https://github.com/dask/distributed/issues/4902
+        gpuci_logger "Clone Distributed"
+        git clone https://github.com/dask/distributed
+
+        gpuci_logger "Run Distributed Tests"
+        py.test --cache-clear -vs distributed/distributed/protocol/tests/test_cupy.py
+        py.test --cache-clear -vs distributed/distributed/protocol/tests/test_numba.py
+        py.test --cache-clear -vs distributed/distributed/protocol/tests/test_rmm.py
+        py.test --cache-clear -vs distributed/distributed/protocol/tests/test_collection_cuda.py
+        py.test --cache-clear -vs distributed/distributed/tests/test_nanny.py
+        py.test --cache-clear -vs distributed/distributed/diagnostics/tests/test_nvml.py
```

Comment:
```suggestion
        pytest --cache-clear -vs distributed/distributed/protocol/tests/test_cupy.py
        pytest --cache-clear -vs distributed/distributed/protocol/tests/test_numba.py
        pytest --cache-clear -vs distributed/distributed/protocol/tests/test_rmm.py
        pytest --cache-clear -vs distributed/distributed/protocol/tests/test_collection_cuda.py
        pytest --cache-clear -vs distributed/distributed/tests/test_nanny.py
        pytest --cache-clear -vs distributed/distributed/diagnostics/tests/test_nvml.py
```

While we're touching these lines, can we update these commands to `pytest` instead of `py.test`? It seems that `py.test` may be deprecated according to the SO discussion below.

https://stackoverflow.com/q/39495429